### PR TITLE
[package] Check control file existence and clarify error messages

### LINKED ISSUE #458: Projects should not require/expect a control file
What are the steps to reproduce this issue?
-------------------------------------------
1. Create a `tool` (currently `iphone/tool`)
2. `cd PROJECT`
3. `rm control`
4. `make TARGET=macosx` 

What happens?
-------------
```text
grep: PROJECT/control: No such file or directory
grep: PROJECT/control: No such file or directory
```

What were you expecting to happen?
----------------------------------
No `grep` errors

Any logs, error output, etc?
----------------------------
```shell
$ ls -a
.        ..       Makefile main.c
$ cat Makefile 
include $(THEOS)/makefiles/common.mk

TOOL_NAME = demo
demo_FILES = main.c

include $(THEOS_MAKE_PATH)/tool.mk
$ cat main.c 
int main() {
	return 0;
}
$ make TARGET=macosx:clang::10.10
grep: PROJECT/control: No such file or directory
grep: PROJECT/control: No such file or directory
> Making all for tool demo…
==> Compiling main.c (x86_64)…
==> Linking tool demo (x86_64)…
==> Generating debug symbols for demo…
==> Merging tool demo…
==> Signing demo…
$ 
```

Any other comments?
-------------------
Everything works fine, just the `grep` errors show up on every `make`. 
This error does not occur when targeting `iphoneos`.

What versions of software are you using?
----------------------------------------
**Operating System:** macOS 10.15.1

**Toolchain Version:** theos/theos @ badb424, Xcode 11.1

**SDK Version:** macosx:10.15

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
1) Changes the *_CAN_PACKAGE check to check for the control/spec file's existence (right now I think it's just checking that the variable's value exists once expanded)

2) Updates the *_CAN_PACKAGE error to note why the control/spec file is required 

Does this close any currently open issues?
------------------------------------------
Resolves #458 

Any relevant logs, error output, etc?
-------------------------------------
```
==> Error: make package requires you to have a control file either in the layout/ directory or in the project root. The control is used to determine info about the package (e.g., name, arch, and version).
make: *** [/home/lightmann/theos/makefiles/package/deb.mk:70: internal-package] Error 1
```

Any other comments?
-------------------
I updated this check for package/*.mk (except none.mk) and have tested the deb and pkgbuild ones.

Where has this been tested?
---------------------------
**Operating System:** …

- Linux (WSL)
- Catalina VM 

**Platform:** …

**Target Platform:** …

- iOS 
- macOS

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: makefiles/package/deb.mk [hunk 1]
```diff
@@ -10,7 +10,11 @@ THEOS_PLATFORM_DEB_COMPRESSION_LEVEL ?= 0
 endif
 
 _THEOS_DEB_PACKAGE_CONTROL_PATH := $(or $(wildcard $(THEOS_PROJECT_DIR)/control),$(wildcard $(THEOS_LAYOUT_DIR)/DEBIAN/control))
-_THEOS_DEB_CAN_PACKAGE := $(if $(_THEOS_DEB_PACKAGE_CONTROL_PATH),$(_THEOS_TRUE),$(_THEOS_FALSE))
+ifneq ("$(wildcard $(_THEOS_DEB_PACKAGE_CONTROL_PATH))","")
```
Comment (kirb) [inline makefiles/package/deb.mk:13]: This can remain `$(if …)` because empty string is equivalent to false.

File: makefiles/package/deb.mk [hunk 2]
```diff
@@ -63,7 +67,7 @@ after-package:: __THEOS_LAST_PACKAGE_FILENAME = $(_THEOS_DEB_PACKAGE_FILENAME)
 
 else # _THEOS_DEB_CAN_PACKAGE == 0
 internal-package::
-	$(ERROR_BEGIN)"$(MAKE) package requires you to have a layout/ directory in the project root, containing the basic package structure, or a control file in the project root describing the package."$(ERROR_END)
+	$(ERROR_BEGIN)"$(MAKE) package requires you to have a control file either in the layout/ directory or in the project root. The control is used to determine info about the package (e.g., name, arch, and version)."$(ERROR_END)
```
Comment (kirb) [inline makefiles/package/deb.mk:70]: We could clarify this further to indicate that it’s layout/DEBIAN/.

**Review by kirb — CHANGES_REQUESTED** [review]
Good thinking, thanks! Just got 2 nitpicky suggestions for you.

File: makefiles/package/deb.mk (hunk 1 above)
Comment (L1ghtmann) [reply makefiles/package/deb.mk]: Wow did I ever overcomplicate things. Sorry, am still kinda new to make syntax. Switched back to the og syntax.

File: makefiles/package/deb.mk (hunk 2 above)
Comment (L1ghtmann) [reply makefiles/package/deb.mk]: Good call. Done!

File: makefiles/package/deb.mk (hunk 1 above)
Comment (kirb) [reply makefiles/package/deb.mk]: That’s fine! Make is an awkward language I definitely don’t expect everyone to know all the tricks to it, totally fine if you write out whatever makes sense to you and I’ll give cleanup hints.

**Review by kirb — APPROVED** [review]
LGTM :+1: Thanks!

**Pull request merged**